specifyweb/frontend/js_src/lib/components/Interactions/InteractionDialog.tsx

```diff
@@ -304,10 +304,7 @@ export function InteractionDialog({
               />
               <div>
                 <Button.Blue
-                  disabled={
-                    catalogNumbers.length === 0 ||
-                    inputRef.current?.validity.valid !== true
-                  }
+                  disabled={catalogNumbers.length === 0}
```

Comment:
The bug occurred because I relied on a ref (`inputRef`) during render. You would think I would know better than to do that

Refs must never be used during render (or at least not without a very good reason). They should only be used inside of hooks. The reason is that changes to refs don't cause re-render, thus a stale value could be replaced

@CarolineDenis your change would make the button be enabled even if there are some invalid catalog numbers. @grantfitzsimmons is that the desired change?

Reply:
I would think all errors should be resolved before proceeding– it would be bad if some catalog numbers were left out because it didn't warn them.